File: src/types/timeline.ts
export type TimelineImportance = 'critical' | 'high' | 'normal' | 'low';

export interface TimelineEvent {
  id: string;
  project_id: string;
  title: string;
  event_date: string | null;
  date_value: string | null;
  date_label: string | null;
  time_value: string | null;
  order_index: number;
  description: string | null;
  location_id: string | null;
  location_name: string | null;
  importance: TimelineImportance | string;
  related_chapter_id: string | null;
  chapter_title: string | null;
  character_ids: string[];
  character_names: string[];
  tags: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateTimelineEventInput {
  project_id: string;
  title: string;
  event_date?: string | null;
  date_value?: string | null;
  date_label?: string | null;
  time_value?: string | null;
  order_index?: number | null;
  description?: string | null;
  location_id?: string | null;
  importance?: TimelineImportance | string | null;
  related_chapter_id?: string | null;
  character_ids?: string[] | null;
  tags?: string | null;
}

export interface UpdateTimelineEventInput {
  title?: string;
  event_date?: string | null;
  date_value?: string | null;
  date_label?: string | null;
  time_value?: string | null;
  order_index?: number | null;
  description?: string | null;
  location_id?: string | null;
  importance?: TimelineImportance | string | null;
  related_chapter_id?: string | null;
  character_ids?: string[] | null;
  tags?: string | null;
}

export interface TimelineFilter {
  character_id?: string | null;
  location_id?: string | null;
  related_chapter_id?: string | null;
  importance?: string | null;
  tag?: string | null;
  search_query?: string | null;
  sort_direction?: 'asc' | 'desc' | null;
}

File: src-tauri/src/models/timeline.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TimelineEvent {
    pub id: String,
    pub project_id: String,
    pub title: String,
    pub event_date: Option<String>,
    pub date_value: Option<String>,
    pub date_label: Option<String>,
    pub time_value: Option<String>,
    pub order_index: i64,
    pub description: Option<String>,
    pub location_id: Option<String>,
    pub location_name: Option<String>,
    pub importance: String,
    pub related_chapter_id: Option<String>,
    pub chapter_title: Option<String>,
    pub character_ids: Vec<String>,
    pub character_names: Vec<String>,
    pub tags: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateTimelineEventInput {
    pub project_id: String,
    pub title: String,
    pub event_date: Option<String>,
    pub date_value: Option<String>,
    pub date_label: Option<String>,
    pub time_value: Option<String>,
    pub order_index: Option<i64>,
    pub description: Option<String>,
    pub location_id: Option<String>,
    pub importance: Option<String>,
    pub related_chapter_id: Option<String>,
    pub character_ids: Option<Vec<String>>,
    pub tags: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateTimelineEventInput {
    pub title: Option<String>,
    pub event_date: Option<String>,
    pub date_value: Option<String>,
    pub date_label: Option<String>,
    pub time_value: Option<String>,
    pub order_index: Option<i64>,
    pub description: Option<String>,
    pub location_id: Option<String>,
    pub importance: Option<String>,
    pub related_chapter_id: Option<String>,
    pub character_ids: Option<Vec<String>>,
    pub tags: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TimelineFilter {
    pub character_id: Option<String>,
    pub location_id: Option<String>,
    pub related_chapter_id: Option<String>,
    pub importance: Option<String>,
    pub tag: Option<String>,
    pub search_query: Option<String>,
    pub sort_direction: Option<String>, // "asc" or "desc"
}
